eth/api.go

```diff
@@ -833,3 +833,43 @@ func (api *ScrollAPI) CalculateRowConsumptionByBlockNumber(ctx context.Context,
 	asyncChecker.Wait()
 	return rawdb.ReadBlockRowConsumption(api.eth.ChainDb(), block.Hash()), checkErr
 }
+
+// SetRollupEventSyncedL1Height sets the synced L1 height for rollup event synchronization
+func (api *ScrollAPI) SetRollupEventSyncedL1Height(height uint64) error {
+	prevHeightPtr := rawdb.ReadRollupEventSyncedL1BlockNumber(api.eth.ChainDb())
+
+	if prevHeightPtr == nil {
+		log.Warn("No previous rollup event synced L1 height found in database")
+		return fmt.Errorf("no previous rollup event synced L1 height found in database")
+	}
+
+	prevHeight := *prevHeightPtr
+	if height >= prevHeight {
+		log.Warn("New rollup event synced L1 height is not lower than previous height", "newHeight", height, "prevHeight", prevHeight)
+		return fmt.Errorf("new rollup event synced L1 height (%d) is not lower than previous height (%d)", height, prevHeight)
+	}
+
+	log.Info("Setting rollup event synced L1 height", "height", height)
+	rawdb.WriteRollupEventSyncedL1BlockNumber(api.eth.ChainDb(), height)
```

Comment:
As I see event sync (with correct L1 endpoint) is idempotent, so this should not break anything. But it would be better if we can test this on a node.

Reply:
tested and the update works fine.